### DISCUSSION TITLE
Scaling a Ruffle Swf in a browser window

### DISCUSSION THREAD

**Discussion by Mark0669** [ruffle-rs/ruffle#3539] [Q&A] [answered]
Is it possible to have a swf scale to the size of a browser window? I use the following code to run, I've tried placing in a container etc but with no success. I'm guessing this is a lack of html knowledge!! Basically is it possible to remove the width and height elements and have the swf scale to the size of the browser window? Any help appreciated! Cheers, Mark

<body>
	 <div id="ruffle"></div>
        <script src="ruffle/ruffle.js"></script>
        <script>
			
var swfobject = {};

swfobject.embedSWF = function(url, cont, width, height){
    var ruffle = window.RufflePlayer.newest(),
        player = Object.assign(document.getElementById(cont).appendChild(ruffle.createPlayer()), {
            width: width,
            height: height,
            style: 'width: ' + width + 'px; height: ' + height + 'px',
        });
    player.config = {
		autoplay: "on",
		unmuteOverlay: "hidden",
		contextMenu: false, // toggles menu for fullscreen
};
    player.load({ url: url });
}

swfobject.embedSWF('iRMwipeout2.swf', 'ruffle', 1000, 800);
        </script>
		
</body>

**Comment by torokati44** [comment]
The [Demo page on ruffle.rs](https://ruffle.rs/demo/) behaves similarly. You could try taking a look at how it's done there.
You can find the sources of that page here: https://github.com/ruffle-rs/ruffle/tree/master/web/packages/demo/www

**Reply by Mark0669** [reply]
Thanks! I have previously looked at that, but trying to extract the right code has proved problematic as the demo script is very heavily geared towards the swf select & drag and drop options. Just trying to make things simplified e.g. single swf embedded on html page, scaleable to window size.

**Reply by torokati44** [reply]
I think the scaling is mostly done by the HTML and CSS parts anyway.

**Reply by Mark0669** [reply]
Absolutely, I'm just struggling to figure out which bits of code I need.

**Reply by nicomicap** [reply]
https://addons.mozilla.org/en-US/firefox/addon/fullscreen-for-anything/ click the addon, then click the game. I don't know an alternative addon for chromium browsers though

**Comment by Mark0669** [comment, marked as answer]
Solved with a simple change 
style: 'width: 100vw; height: 10vh',
